module/mempool/stdmap/backdata/heropool/pool.go

```diff
@@ -0,0 +1,300 @@
+package heropool
+
+import (
+	"math/rand"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type EjectionMode string
+
+const (
+	RandomEjection = EjectionMode("random-ejection")
+	LRUEjection    = EjectionMode("lru-ejection")
+)
+
+// number of entries we try to eject before giving up and ejecting LRU
+const maximumRandomTrials = 10
+
+// EIndex is data type representing an entity index in Pool.
+type EIndex uint32
+
+// cachedEntity represents a cached entity that is maintained as a double linked list node.
+type cachedEntity struct {
```

Comment:
Consider renaming `cachedEntity` to e.g. `poolEntity` or `poolItem`. Why? This particular implementation of heropool is being used for caching, but it is already relatively abstract and it could be used in future for other hero implementations of algorithms which do not feature caching, e.g. managing a pool of 100k TCP connections, or whatever where no caching is involved...

Reply:
Addressed in https://github.com/onflow/flow-go/pull/1778/commits/bff8e193827d843b90b97bedfcf3b1b70b479cf0